markets/spot-market/contracts/storage/Fee.sol

```diff
@@ -213,10 +216,23 @@ library Fee {
         }
     }
 
+    function collectFees(uint128 marketId, uint totalFees) internal returns (uint feesCollected) {
+        IFeeCollector feeCollector = load(marketId).feeCollector;
+        if (address(feeCollector) != address(0)) {
+            SpotMarketFactory.Data storage store = SpotMarketFactory.load();
+
+            store.usdToken.approve(address(feeCollector), totalFees);
+            feesCollected = feeCollector.collectFees(totalFees);
+            store.usdToken.approve(address(feeCollector), 0);
+
+            store.depositToMarketManager(marketId, msg.sender, totalFees - feesCollected);
+        }
+    }
```

Comment:
I imagined this function would always end with `depositToMarketManager` (or this functionality would exist in the `depositToMarketManager` function). i.e. we have a 'wrapper' around `depositToMarketManager` (essentially "improve market PnL"), where a custom contract has an opportunity to intercept some of profit before it goes back into the system.

Reply:
yea makes sense.  i added `processFees` that can be called, and it will calculate fees, run through fee collector then deposit the rest into market manager all in that call.